packages/docsify-server-renderer/index.js

```diff
@@ -118,20 +105,65 @@ export default class Renderer {
         path = cover === true ? '_coverpage.md' : cover;
       }
 
-      const coverFile = this._getPath(resolve(url, `./${path}`));
-
-      this._renderHtml('cover', await this._render(coverFile), 'cover');
+      const coverFile = this._getPath(path);
+      this._renderHtml('cover', await this._render(coverFile, 'cover'));
     }
 
-    const html = this.isRemoteUrl ? DOMPurify.sanitize(this.html) : this.html;
-    this.html = this.template;
+    let html = this.isExternal ? DOMPurify.sanitize(this.html) : this.html;
     return html;
   }
 
-  _renderHtml(match, content) {
-    this.html = this.html.replace(new RegExp(`<!--${match}-->`, 'g'), content);
+  _getWebConfig() {
+    let config = Object.assign({}, this.config);
+    config.routerMode = 'abstract';
```

Comment:
why assigning router to abstract ? is it for the default ?

Reply:
Since the page is rendered in the server-side, there is no need for web page javascripts to deal sidebar links as 'history' or 'hash' mode, it is directly an `<a>` link. Thus, I uses 'abstract' as the mode to avoid any js `on-` hooks.

Reply:
cc @docsifyjs/reviewers

Reply:
@anikethsaha Sorry that I found the e2e tests goes through `history` and `hash` modes. I'll change it by blocking `on-` hooks in `history` and `hash` modes when it is in SSR mode.

Reply:
@anikethsaha 
I found that e2e tests runs through both hash and history modes. Thus, I am going to change this as following:
1. Since SSR only goes an `<a>` link, and no need for `onchange` actions, I will use `abstract` as routerMode in injected config.
2. Since SSR is in node, which does not have `window` object, I will use default `getBasePath`, `getCurrentPath` and `normalize` methods from `History` class.
3. Since `parse` and `toURL` methods are required for generating links, I will use corresponding methods from either `HashHistory` or `HTML5History` class'es prototype.

**Note**: SSR config is slightly different from injected config. Injected one has routerMode 'abstract' to avoid `onchange` hooks. SSR config has routerMode 'hash' or 'history' to make sure the links are rendered correctly.

Reply:
We are not doing e2e for SSR I suppose, Also I would recommend to not change the current tests as it is being completely re-written https://github.com/docsifyjs/docsify/pull/1276